### PR TITLE
LA-137 Received Share List

### PR DESCRIPTION
**Pull request by hoangdat**
IMG_0084

### CODE REVIEW

File: testshared/pubspec.yaml [hunk 1]
```diff
@@ -21,6 +21,8 @@ dependencies:
   #rxdart
   rxdart: 0.24.1
 
+  dartz: ^0.9.2
```
Comment (manhduc-nguyen) [inline testshared/pubspec.yaml:24]: we still using `^` ?

**Review by manhduc-nguyen — APPROVED** [review]

File: domain/lib/domain.dart [hunk 2]
```diff
@@ -1,108 +1,105 @@
 library domain;
 
+// errorcode
+export 'src/errorcode/business_error_code.dart';
 // extension
 export 'src/extension/email_validator_string_extension.dart';
-
-// viewState
-export 'src/state/failure.dart';
```
[the remark this answers is not in the document]
Comment (imGok) [reply domain/lib/domain.dart:7]: holy, we gonna suffer

File: lib/presentation/redux/states/received_share_state.dart [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    return ReceivedShareState(Right(IdleState()), receivedList, selectMode);
+  }
+
+  @override
+  LinShareState sendViewState({Either<Failure, Success> viewState}) {
+    throw ReceivedShareState(viewState, receivedList, selectMode);
+  }
+
+  @override
+  LinShareState startLoadingState() {
```
Comment (imGok) [inline lib/presentation/redux/states/received_share_state.dart:67]: Why not returning ReceivedShareState for this state ?

File: domain/lib/domain.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (manhduc-nguyen) [reply domain/lib/domain.dart:7]: Pray the lord

**Comment by manhduc-nguyen** [comment]
@imGok @hoangdat pls review this guys.

**Comment by hoangdat** [comment]
Thank Duc so much! LGTM! Approved

**Review by imGok — APPROVED** [review]

**Pull request merged**